lms/djangoapps/courseware/courses.py

```diff
@@ -764,6 +773,38 @@ def _ora_assessment_to_assignment(
     )
 
 
+def get_assignments_grades(user, course_id, cache_timeout):
+    """
+    Calculate the progress of the assignment for the user in the course.
+
+    Arguments:
+        user (User): Django User object.
+        course_id (CourseLocator): The course key.
+        cache_timeout (int): Cache timeout in seconds
+    Returns:
+        list (ReadSubsectionGrade, ZeroSubsectionGrade): The list with assignments grades.
+    """
+    is_staff = bool(has_access(user, 'staff', course_id))
+
+    try:
+        cache_key = f'course_block_structure_{str(course_id)}_{user.id}'
```

Comment:
I believe we should use `course_version` attribute, to make sure that cache is invalidated with each new published course version
Please see https://github.com/raccoongang/edx-platform/blob/master/lms/djangoapps/course_home_api/outline/views.py#L456 for similar caching approach.

Reply:
Yes, added cache invalidation with course version